In [4]:
import pandas as pd

In [5]:
# Load in the dataframe from project part 1 as basics:
df_2000 = pd.read_csv('Data/final_tmdb_data_2000.csv.gz', compression = 'gzip')

df_2001 = pd.read_csv('Data/final_tmdb_data_2001.csv.gz', compression = 'gzip')

In [6]:
df = pd.concat([df_2000, df_2001])

In [7]:
df.head()

imdb_id  adult                     backdrop_path belongs_to_collection  \
0          0    NaN                               NaN                   NaN   
1  tt0113026    0.0  /vMFs7nw6P0bIV1jDsQpxAieAVnH.jpg                   NaN   
2  tt0113092    0.0                               NaN                   NaN   
3  tt0116391    0.0                               NaN                   NaN   
4  tt0118694    0.0  /n4GJFGzsc7NinI1VeGDXIcQjtU2.jpg                   NaN   

       budget                                             genres homepage  \
0         NaN                                                NaN      NaN   
1  10000000.0  [{'id': 35, 'name': 'Comedy'}, {'id': 10402, '...      NaN   
2         0.0           [{'id': 878, 'name': 'Science Fiction'}]      NaN   
3         0.0  [{'id': 18, 'name': 'Drama'}, {'id': 28, 'name...      NaN   
4    150000.0  [{'id': 18, 'name': 'Drama'}, {'id': 10749, 'n...      NaN   

         id original_language   original_title  ...     revenue  runtime  \
0       NaN               NaN              NaN  ...         NaN      NaN   
1   62127.0                en  The Fantasticks  ...         0.0     86.0   
2  110977.0                en    For the Cause  ...         0.0    100.0   
3  442869.0                hi             Gang  ...         0.0    152.0   
4     843.0                cn             花樣年華  ...  14204632.0     99.0   

                                    spoken_languages    status  \
0                                                NaN       NaN   
1  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
2  [{'english_name': 'English', 'iso_639_1': 'en'...  Released   
3  [{'english_name': 'Hindi', 'iso_639_1': 'hi', ...  Released   
4  [{'english_name': 'Cantonese', 'iso_639_1': 'c...  Released   

                                             tagline                 title  \
0                                                NaN                   NaN   
1      Try to remember the first time magic happened       The Fantasticks   
2       The ultimate showdown on a forbidden planet.         For the Cause   
3                                                NaN                  Gang   
4  Feel the heat, keep the feeling burning, let t...  In the Mood for Love   

   video  vote_average vote_count certification  
0    NaN           NaN        NaN           NaN  
1    0.0         5.500       22.0           NaN  
2    0.0         5.450       10.0           NaN  
3    0.0         4.000        1.0           NaN  
4    0.0         8.111     2305.0            PG  

[5 rows x 26 columns]

In [8]:
df['budget'].value_counts

<bound method IndexOpsMixin.value_counts of 0              NaN
1       10000000.0
2              0.0
3              0.0
4         150000.0
           ...    
1346           0.0
1347           0.0
1348           0.0
1349           0.0
1350           0.0
Name: budget, Length: 2608, dtype: float64>

In [9]:
# selecting rows based on condition
df = df[(df['budget'] > 0) &
		df['revenue'] > 0]
	

### How many movies are there in each of the certification categories (G/PG/PG-13/R)?

In [10]:
df['certification'].value_counts()

R        143
PG-13    117
PG        30
G         13
NR         8
Name: certification, dtype: int64

G = 13
PG = 30
PG-13 = 117
R = 143

### What is the average revenue per certification category?


In [11]:
df['revenue'].mean()

77549660.32869081

$77,549,660.32

### What is the average budget per certification category?

In [12]:
df['budget'].mean()

33921943.19220056

$33,921,943.29